src/lib/DefaultRuleManager.ts

```diff
@@ -98,14 +98,36 @@ export class DefaultRuleManager implements RuleManager {
 	private async unpackTargets(engine: RuleEngine, targets: string[]): Promise<RuleTarget[]> {
 		const ruleTargets: RuleTarget[] = [];
 		// Ask engines for their desired target patterns.
-		const targetPatterns: string[] = await engine.getTargetPatterns();
-		assert(targetPatterns);
-		const pm = new PathMatcher(targetPatterns);
-		for (const target of targets) {
-			const fileExists = await this.fileHandler.exists(target);
+		const engineTargets = await engine.getTargetPatterns();
+		assert(engineTargets);
+		// We also need to do a bit of processing on the patterns we were given.
+		const positivePatterns = [];
+		const negativePatterns = [];
+		targets.forEach((t) => {
+			if (t.startsWith('!**') || t.startsWith('!/')) {
+				// If a negative glob starts with a ** or /, it's an absolute path and we can just add it to our array.
+				negativePatterns.push(t);
+			} else if (t.startsWith('!')) {
+				// If relative negative paths start with dots (e.g., ./ or ../../), we should use those to resolve to an
+				// absolute path. Otherwise, we should just use the current directory to resolve to an absolute path.
+				const regexResult = /(\.{1,2}\/)+/.exec(t);
+				const pathPrefix = regexResult ? regexResult[0] : '.';
+				const resolvedPrefix = path.resolve(pathPrefix);
+				const globStartPoint = (regexResult ? pathPrefix.length : 0) + 1;
+				negativePatterns.push(`!${resolvedPrefix}/${t.slice(globStartPoint)}`);
```

Comment:
The tertiary comparisons make this a little hard to read. Grouping the regexResult tertiarties or breaking them out might make it clearer. The thing that confused me at first was trying to figure out how `regexResult` and `pathPrefix` influence the value of `globsStartPoint`.

```
const pathPrefix = regexResult ? regexResult[0] : '.';
const globStartPoint = (regexResult ? pathPrefix.length : 0) + 1;
const resolvedPrefix = path.resolve(pathPrefix);
negativePatterns.push(`!${resolvedPrefix}/${t.slice(globStartPoint)}`);
```
or
```
let resolvedPrefix;
let globStartPoint;
if (regexResult) {
    resolvedPrefix = path.resolve(regexResult[0]);
    globStartPoint = pathPrefix.length + 1;
} else {
    resolvedPrefix = path.resolve('.');
    globStartPoint = 1;
}
negativePatterns.push(`!${resolvedPrefix}/${t.slice(globStartPoint)}`);
```